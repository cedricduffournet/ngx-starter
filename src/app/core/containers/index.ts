export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
