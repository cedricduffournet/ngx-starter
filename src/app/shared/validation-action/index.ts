export * from './validation-action.module';
