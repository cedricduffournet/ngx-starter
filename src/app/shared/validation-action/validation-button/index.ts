export * from './validation-button.component';
