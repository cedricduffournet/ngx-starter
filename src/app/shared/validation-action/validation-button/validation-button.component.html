<div
  class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
  [ngClass]="formGroupClass"
>
  <button
    *ngIf="labelCancel !== '' && !confirmDisabled"
    type="button"
    (click)="onCancel()"
    class="btn btn-cancel btn-lg"
    ngClass="btn-{{ classCancel }}"
  >
    {{ labelCancel }}
  </button>
  <button
    *ngIf="labelConfirm !== ''"
    [type]="typeConfirm"
    (click)="onConfirm()"
    class="btn btn-confirm btn-lg"
    [disabled]="confirmDisabled"
    ngClass="btn-{{ classConfirm }}"
  >
    <span
      class="spinner-border spinner-border-sm align-baseline"
      style="font-size: 1rem;"
      *ngIf="confirmDisabled"
    ></span>
    <span *ngIf="confirmDisabled">&nbsp;{{ 'LOADING' | translate }}</span>
    <span *ngIf="!confirmDisabled">{{ labelConfirm }}</span>
  </button>
</div>
