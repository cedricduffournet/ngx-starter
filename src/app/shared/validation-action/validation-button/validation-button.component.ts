import {
  Component,
  ChangeDetectionStrategy,
  Input,
  Output,
  EventEmitter
} from '@angular/core';

@Component({
  selector: 'app-validation-button',
  templateUrl: './validation-button.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ValidationButtonComponent {
  @Input() labelConfirm = '';
  @Input() labelCancel = '';
  @Input() typeConfirm = 'submit';
  @Input() classConfirm = 'primary';
  @Input() classCancel = 'light';
  @Input() formGroupClass = '';
  @Input() confirmDisabled = false;
  @Output() confirm = new EventEmitter();
  @Output() cancel = new EventEmitter();

  onConfirm() {
    this.confirm.emit({});
  }

  onCancel() {
    this.cancel.emit({});
  }
}
