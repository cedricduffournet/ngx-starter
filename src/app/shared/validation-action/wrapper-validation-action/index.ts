export * from './wrapper-validation-action.component';
