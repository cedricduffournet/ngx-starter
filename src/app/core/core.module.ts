import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
