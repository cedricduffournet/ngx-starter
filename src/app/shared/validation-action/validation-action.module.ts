import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslateModule } from '@ngx-translate/core';

import { ValidationButtonComponent } from '@app/shared/validation-action/validation-button';
import { WrapperValidationActionComponent } from '@app/shared/validation-action/wrapper-validation-action';

@NgModule({
  imports: [CommonModule, TranslateModule],
  exports: [ValidationButtonComponent, WrapperValidationActionComponent],
  declarations: [ValidationButtonComponent, WrapperValidationActionComponent]
})
export class ValidationActionModule {}
