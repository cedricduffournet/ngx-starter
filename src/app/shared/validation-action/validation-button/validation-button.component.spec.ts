import { TestBed, ComponentFixture } from '@angular/core/testing';

import { TranslateModule } from '@ngx-translate/core';
import { ValidationButtonComponent } from '@app/shared/validation-action/validation-button';

describe('ValidationButtonComponent', () => {
  let fixture: ComponentFixture<ValidationButtonComponent>;
  let component: ValidationButtonComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot()],
      declarations: [ValidationButtonComponent]
    });
    fixture = TestBed.createComponent(ValidationButtonComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be created with cancel and confirm button', () => {
    component.labelCancel = 'cancel';
    component.labelConfirm = 'confirm';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('root div shoud have class', () => {
    component.formGroupClass = 'testclass';

    fixture.detectChanges();

    const div = fixture.nativeElement.querySelector(
      '.button-validation-action.testclass'
    );
    expect(div).toBeTruthy();
  });

  it('cancel button should have class', () => {
    component.labelCancel = 'cancel';
    component.classCancel = 'danger';

    fixture.detectChanges();

    const button = fixture.nativeElement.querySelector(
      'button.btn-cancel.btn-danger'
    );
    expect(button).toBeTruthy();
  });

  it('confirm button should have class', () => {
    component.labelConfirm = 'confirm';
    component.classConfirm = 'success';

    fixture.detectChanges();

    const button = fixture.nativeElement.querySelector(
      '.btn-confirm.btn-success'
    );
    expect(button).toBeTruthy();
  });

  it('confirm button should be type button', () => {
    component.labelConfirm = 'confirm';
    component.typeConfirm = 'button';

    fixture.detectChanges();

    const button = fixture.nativeElement.querySelector('.btn-confirm');
    expect(button.type).toBe('button');
  });

  it('confirm button should be disabled', () => {
    component.labelConfirm = 'confirm';
    component.confirmDisabled = true;

    fixture.detectChanges();

    const button = fixture.nativeElement.querySelector('.btn-confirm');
    expect(button.disabled).toBeTruthy();
  });

  it('cancel button should not be displayed when confirm button is disabled', () => {
    component.labelConfirm = 'confirm';
    component.labelCancel = 'cancel';
    component.confirmDisabled = true;

    fixture.detectChanges();

    const button = fixture.nativeElement.querySelector('.btn-cancel');
    expect(button).toBeFalsy();
  });

  it('spinner should be displayed when confirm is disabled', () => {
    component.labelConfirm = 'confirm';
    component.confirmDisabled = true;

    fixture.detectChanges();

    const spinner = fixture.nativeElement.querySelector('.spinner-border');
    expect(spinner).toBeTruthy();
  });

  it('confirm event should be emitted when confirm button has been clicked', () => {
    spyOn(component.confirm, 'emit');
    component.labelConfirm = 'confirm';
    fixture.detectChanges();
    const confirm = fixture.nativeElement.querySelector('.btn-confirm');
    confirm.click();
    expect(component.confirm.emit).toHaveBeenCalled();
  });

  it('cancel event should be emitted when cancel button has been clicked', () => {
    spyOn(component.cancel, 'emit');
    component.labelCancel = 'confirm';
    fixture.detectChanges();
    const cancel = fixture.nativeElement.querySelector('.btn-cancel');
    cancel.click();
    expect(component.cancel.emit).toHaveBeenCalled();
  });
});
