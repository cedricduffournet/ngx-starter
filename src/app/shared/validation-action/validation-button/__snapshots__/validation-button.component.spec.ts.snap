// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationButtonComponent should be created 1`] = `
<app-validation-button
  cancel={[Function EventEmitter]}
  classCancel={[Function String]}
  classConfirm={[Function String]}
  confirm={[Function EventEmitter]}
  confirmDisabled="false"
  formGroupClass=""
  labelCancel=""
  labelConfirm=""
  typeConfirm={[Function String]}
>
  <div
    class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
    ng-reflect-klass="button-validation-action text-"
    ng-reflect-ng-class=""
  >
    
    
  </div>
</app-validation-button>
`;

exports[`ValidationButtonComponent should be created with cancel and confirm button 1`] = `
<app-validation-button
  cancel={[Function EventEmitter]}
  classCancel={[Function String]}
  classConfirm={[Function String]}
  confirm={[Function EventEmitter]}
  confirmDisabled="false"
  formGroupClass=""
  labelCancel={[Function String]}
  labelConfirm={[Function String]}
  typeConfirm={[Function String]}
>
  <div
    class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
    ng-reflect-klass="button-validation-action text-"
    ng-reflect-ng-class=""
  >
    
    <button
      class="btn btn-cancel btn-lg btn-light"
      ng-reflect-klass="btn btn-cancel btn-lg"
      ng-reflect-ng-class="btn-light"
      type="button"
    >
       cancel 
    </button>
    
    <button
      class="btn btn-confirm btn-lg btn-primary"
      ng-reflect-klass="btn btn-confirm btn-lg"
      ng-reflect-ng-class="btn-primary"
      type="submit"
    >
      
      
      
      <span>
        confirm
      </span>
    </button>
  </div>
</app-validation-button>
`;
